From fc0eedd0407ace352c3bab08e3e75b52386cad7e Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Wed, 13 Sep 2023 19:22:51 +0900
Subject: [PATCH] makefile.in: dont compress loaddefs files

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index fdd9353e2..d32f5444b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -759,7 +759,7 @@ install-arch-indep:
 	[ -z "${GZIP_PROG}" ] || { \
 	  echo "Compressing *.el etc. ..." && \
 	  cd "$(DESTDIR)${lispdir}" && \
-	  for f in `find . -name "*.elc" -print | sed 's/.elc$$/.el/'`; do \
+	  for f in `find . -name "*.elc" ! -name "*loaddefs.elc" -print | sed 's/.elc$$/.el/'`; do \
 	    ${GZIP_PROG} -9n "$$f"; \
 	  done; \
 	  ${GZIP_PROG} -9n "../etc/publicsuffix.txt"; \
-- 
2.42.0

