From c7513e8220563d65cb40d5313196277875c0fa26 Mon Sep 17 00:00:00 2001
From: Kien Nguyen <kien.n.quang@gmail.com>
Date: Fri, 23 Aug 2024 04:51:47 -0700
Subject: [PATCH] Stop using legacy -ffat-lto-objects option in LTO builds

---
 configure.ac | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index ac5660bbd..1323844f4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1974,13 +1974,6 @@ AC_DEFUN
 	 # command, so plugin name is appended to ARFLAGS.
 	 ARFLAGS="cru --plugin $GOLD_PLUGIN"
 	 RANLIB="$RANLIB --plugin $GOLD_PLUGIN"
-      else
-        dnl The following is needed for GCC 4.9.0.  The GCC 4.9.0 release notes
-        dnl suggest that instead of -ffat-lto-objects we should use gcc-ar and
-        dnl gcc-ranlib in place of ar and ranlib, but gcc-ar makes /usr/bin/ar
-        dnl dump core on Fedora 20, so play it safe for now.
-        gl_COMPILER_OPTION_IF([-ffat-lto-objects],
-          [CFLAGS="$CFLAGS -ffat-lto-objects"])
       fi
    fi
 fi)
-- 
2.46.0.vfs.0.0

